File: CarWad/Queue.cs
using System;

namespace carWash
{
    public class Queue<T>
    {
        private Node<T>? head;
        private Node<T>? tail;
        private int size;

        public int Size => size;
        public bool IsEmpty => size == 0;
        public Queue()
        {
            head = null;
            tail = null;
            size = 0;
        }

        public void Add(T data)
        {
            Node<T> newNode = new Node<T>(data);
            if(IsEmpty)
            {
                head = newNode;
            }
            else
            {
                tail.Next = newNode;
            }
            tail = newNode;
            size++;
        }

        public T GetValue()
            {
            if (IsEmpty )
            {
                throw new InvalidOperationException("La cola está vacía");
            } 
            return head.Data;
            }
        public T pullValue()
        {
            if (IsEmpty)
            {
                throw new InvalidOperationException("La cola está vacía");
            } 

            T aux = head.Data;
            head = head.Next;
            if (head == null)
            {
                tail = null;
            }
            size--;
            return aux;
        }
        public void clear()
        {
            head = null;
            tail = null;
            size = 0;
        }
    }
}

File: List in java/src/Queue.java
public class Queue<T> {
    private Node<T> head;
    private Node<T> tail;


    public Queue() {
        head = null;
        tail = null;
    }

    public void add(T t){
        Node<T> newNode = new Node<T>(t);
        if (head==null) {
            head = newNode;
            tail = newNode;
        }

        tail.setNext(newNode);
        tail = newNode;
    }
    public T get(){
        if (head == null) {
            throw new NullPointerException("No elemens in the Queue");
        }
        T auxT = head.getData();
        
        head = (head == tail) ? null : head.getNext();
        return auxT;
    }
    public T peak(){
        return head.getData();
    }
    public boolean isEmpty(){
        return head==null;
    }

}
